VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,850 @@
+   can be replaced by `inside`.
+
+:-1:
+```systemverilog
+constraint con {
+  foreach (data[i]) {
+    a != data[i];
+  }
+}
+```
+
+:+1:
+```systemverilog
+constraint con {
+  !(a inside {data});
+}
+```
+
+2. If using loops in constraints is necessary, avoid doing calculations inside
+   the loops.
+3. Whenever possible, use bitmasking operations over modulus operations, as
+   bitmasking operations are much faster.
+4. If a constraint becomes too complex, it is highly recommended to split the
+   constraint and separately randomize all relevant variables.
+5. When randomizing an array of objects, it is generally recommended to not
+   directly randomize the array, but to iterate through it and directly
+   randomize each object to see better simulation performance.
+
+
+### Enums
+
+Enum names should be written in `lower_snake_case`, and should include the block
+name for clarity.
+The enumerated values should be written in `UpperCamelCase` and should include
+the name of the enum for clarity.
+
+```systemverilog
+typedef enum bit {
+  UartInterruptFrameErr,
+  UartInterruptRxBufferFull,
+  ...
+} uart_interrupte_e;
+```
+
+
+### Interfaces, Clocking Blocks, Modports
+
+1. Do not use the `program` construct.
+2. Clocks and reset must be generated in SystemVerilog modules, interfaces,
+or in the UVM component hierarchy.
+Do not do this generation in `uvm_object` derived classes.
+3. Use clocking blocks in a SystemVerilog interface to sample and drive
+synchronous DUT interfaces.
+4. It is recommended to implement a reusable interface scheme for block level
+testbenches that will be included in integration/system level testbenches.
+
+
+### Loop Operators
+
+It is highly recommended to use `foreach` loops as often as possible over any
+equivalent loop operators, as they are much more concise and less prone to any
+subtle errors.
+
+
+### Code Within Asserts
+
+For any randomization that is required, you must use the macros specified
+in the [Randomization Macros section](#macro-usage), do not use `assert()` to check
+manual randomization calls, use the provided macros instead.
+More broadly, do not place any expression with side effects into an
+`assert()` statement.
+
+:-1:
+```systemverilog
+// Do not assert the output of randomize() calls
+// 1.
+ret = item.randomize() with {<constraints>};
+assert(ret);
+// 2.
+assert(randomize(<variable>) with {<constraints>});
+// 3.
+assert(item.randomize() with {<constraints>});
+// 4.
+assert(function_that_has_side_effects());
+```
+
+### Wait and Fork
+
+Always put `wait fork` and `disable fork` constructs inside of an
+`isolation fork...join` block to avoid erroneous waiting.
+It is recommended to use the `DV_SPINWAIT` macro whenever possible,
+[as described here](#macro-usage)
+
+
+### Void Casts
+
+Do not void cast any function/task calls that have useful return values,
+EXCEPT the following:
+* System functions
+* RAL `predict()` calls
+* FIFO/Queue methods
```

Comment:
maybe align to FIFO or fifo in the whole document for consistency

Reply:
Good point, I changed this usage to lowercase to match the rest of the document.